File: src/client/java/com/somefrills/config/about/GuiElementButton.java
package com.somefrills.config.about;

import io.github.notenoughupdates.moulconfig.common.RenderContext;
import io.github.notenoughupdates.moulconfig.common.text.StructuredText;

import java.awt.*;

public class GuiElementButton {

    public static final int HEIGHT = 18 + 5;
    public int width = -1;
    private String text = "";

    public GuiElementButton() {
    }

    public int getWidth(RenderContext context) {
        var fr = context.getMinecraft().getDefaultFontRenderer();
        return fr.getStringWidth(text) + 10;
    }

    public void render(RenderContext context, int x, int y) {
        context.drawColoredRect(x, y, x + width, y + 18, Color.WHITE.getRGB());
        context.drawColoredRect(x + 1, y + 1, x + width - 1, y + 18 - 1, Color.BLACK.getRGB());
        var fr = context.getMinecraft().getDefaultFontRenderer();
        context.drawString(fr, StructuredText.of(text), x + 5, y + 5, -1, true);
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }
}

File: src/main/kotlin/com/somefrills/config/about/GuiElementButton.kt
package com.somefrills.config.about

import io.github.notenoughupdates.moulconfig.common.RenderContext
import io.github.notenoughupdates.moulconfig.common.text.StructuredText
import java.awt.Color

class GuiElementButton {
    var width: Int = -1
    var text: String = ""

    fun getWidth(context: RenderContext): Int {
        val fr = context.minecraft.defaultFontRenderer
        return fr.getStringWidth(text) + 10
    }

    fun render(context: RenderContext, x: Int, y: Int) {
        context.drawColoredRect(
            x.toFloat(),
            y.toFloat(),
            (x + width).toFloat(),
            (y + 18).toFloat(),
            Color.WHITE.rgb
        )
        context.drawColoredRect(
            (x + 1).toFloat(),
            (y + 1).toFloat(),
            (x + width - 1).toFloat(),
            (y + 18 - 1).toFloat(),
            Color.BLACK.rgb
        )
        val fr = context.minecraft.defaultFontRenderer
        context.drawString(fr, StructuredText.of(text), x + 5, y + 5, -1, true)
    }

    companion object {
        const val HEIGHT: Int = 18 + 5
    }
}

